File: src/Atlasd/Battlenet/Exceptions/GameProtocolViolationException.cs
using System;

namespace Atlasd.Battlenet.Exceptions
{
    class GameProtocolViolationException : GameProtocolException
    {
        public GameProtocolViolationException(ClientState client) : base(client) { }
        public GameProtocolViolationException(ClientState client, string message) : base(client, message) { }
        public GameProtocolViolationException(ClientState client, string message, Exception innerException) : base(client, message, innerException) { }
    }
}

File: src/AtlasV/Battlenet/Exceptions/GameProtocolViolationException.vb
Imports System

Namespace AtlasV.Battlenet.Exceptions
    Class GameProtocolViolationException
        Inherits GameProtocolException

        Public Sub New(ByVal varClient As ClientState)
            MyBase.New(varClient)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String)
            MyBase.New(varClient, varMessage)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(varClient, varMessage, varInnerException)
        End Sub
    End Class
End Namespace
